### PR TITLE
Support sending challenges with a link 

### PR DESCRIPTION
**Pull request by karamnaser**
ther was some issue with code formation had to remove my previous PR
videos link related to issue:
https://streamable.com/l1p99k
https://streamable.com/hn0x32
https://streamable.com/t641dw
in this PR :
[ x ]fixed my code text formation
[ x ]created  popuplisener and added to client in this.socket.onmessege function
[ x ]changed promise to call back in onteamselect
[ x ]changed alert to popups 
[ x ]used Storage.prefs instead of localstoreg 
[ x ]nested all my function inside ManeManuRoom class

### CODE REVIEW

File: js/client-mainmenu.js [hunk 1]
```diff
@@ -81,6 +81,18 @@
...
+						if(!$('.challenge').length){
+							self.generateChallengeForm(self)
+						}
+				}
+				else{
+					window.app.addPopupMessage("you need to log in first to generate link");
+				}
+				});
+				self.createLoginFormObserver(self);
+				window.Popuplesiner=self.Popuplesiner;
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:95]: ?_?
What's this?

File: js/client-mainmenu.js [hunk 2]
```diff
@@ -114,7 +126,131 @@
...
+					$('body').on("login_event",function(){
+					self.onUserLogIn(self);
+				})
+				$('body').trigger("login_event");
+				}
+			}
+			});
+			observer.observe(document.activeElement, config);
+			},
+		onUserLogIn:function(self){
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:145]: Run eslint please.

File: js/client-mainmenu.js [hunk 3]
```diff
@@ -114,7 +126,131 @@
...
+			$('#accept-challenge').html("processing request...");
+				if(window.Storage.prefs("team")!="random"){
+					app.sendTeam(window.Storage.teams[$('#select-team .select.teamselect').val()]);
+					app.send('/vtm ' + $('.select.formatselect').val());
+					self.validateChallengeAndSend(self,{userId:challengerName,massage:window.Storage.prefs("errormassege")});
+				}
+				else{
+					self.validateChallengeAndSend(self,{userId:challengerName,massage:"Your team is valid"});
+				}
+		},
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:171]: Another example block that needs eslint.

File: js/client-mainmenu.js [hunk 4]
```diff
@@ -114,7 +126,131 @@
...
+			$('#generate-link').html("generate link");
+			}
+			else{
+				$('.link').remove();
+				$('#generate-link').html("generate link");
+				window.app.addPopupMessage(window.Storage.prefs("errormassege"));
+				
+			}
+			},2000);
+			});
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:225]: Another example block that needs eslint.

File: js/client-mainmenu.js [hunk 5]
```diff
@@ -114,7 +126,131 @@
...
+		generateLink:function(challengeData){
+			if(!$('.link').length){
+			let link="https://play.pokemonshowdown.com?challenge-"+challengeData.userId+"-"+challengeData.format+"-"+challengeData.teamIndex;
+			$('.link-generatore').append("<p class=link>"+link+"</p>");
+			}
+			else{
+				let link="https://play.pokemonshowdown.com?challenge-"+challengeData.userId+"-"+challengeData.format+"-"+challengeData.teamIndex;
+				$('.link').html("<p class=link>"+link+"</p>")
+			}
+		},
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:194]: Another example block that needs eslint.

File: js/client-mainmenu.js [hunk 6]
```diff
@@ -114,7 +126,131 @@
...
+		generateChallengeForm:function(self){
+			var buf = '<div class="challenge"><form class="battleform"><p>Challenge link generatore</p>';
+			buf += '<p><label class="label">Format:</label>' + self.renderFormats("gen8randombattle") + '</p>';
+			buf += '<p><label class="label">Team:</label>' + self.renderTeams("gen8randombattle") + '</p>';
+			buf += '<p><label class="checkbox"><input type="checkbox" name="private" ' + (Storage.prefs('disallowspectators') ? 'checked' : '') + ' /> <abbr title="You can still invite spectators by giving them the URL or using the /invite command">Don\'t allow spectators</abbr></label></p>';
+			buf +='<button type="button" id=generate-link>generate link</button>';
+			buf +='<button type="button" id=close-challenge>close</button></form></div>'
+			$('.pmbox').append(`<div class=challenge-container>${buf}</div>`);
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:202]:
```suggestion
			$('.pmbox').append('<div class=challenge-container>' + buf + '</div>');
```

File: js/client-mainmenu.js [hunk 7]
```diff
@@ -114,7 +126,131 @@
...
+				
+			}
+			},2000);
+			});
+
+			$('#close-challenge').click(function(){
+				$('.challenge-container').empty();
+			})
+		},
+		sendChallenge:function(challengeDataObject){
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:231]: Why not just use `userid, format` args here?

File: js/client-mainmenu.js [hunk 8]
```diff
@@ -114,7 +126,131 @@
...
+			app.send('/challenge ' + challengeDataObject.userId + ', ' + challengeDataObject.format);
+		},
+		Popuplesiner:function(msg){
+			if(msg.includes("popup")){
+				window.Storage.prefs("errormassege");
+			}
+		},
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:238]: What's this? It doesn't seem to do anything.

File: js/client-mainmenu.js [hunk 9]
```diff
@@ -1487,5 +1623,21 @@
...
+			setTimeout(function(){
+			//changing url without refreshing page, ther was issue of losing data when refreshing page 
+			window.history.replaceState(null,"",'https://play.pokemonshowdown.com/');
+			//log in user before sending challenge
+			new window.UserPopup({name:''}).login();
+			},1000);
+			data=window.location.search.split("-");
+			userId=data[1];
+			format=data[2];
+			window.Storage.prefs("challenger",userId);
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:1638]: Another example block that needs eslint.

**Comment by mia-pi-git** [comment]
In the future, please just update your old PR instead of opening a new one.

For the use of those looking - [the previous PR is here](https://github.com/smogon/pokemon-showdown-client/pull/1969#discussion_r864520541).

File: js/client-mainmenu.js (hunk 2 above)
Comment (karamnaser) [reply js/client-mainmenu.js:145]: ther was issue when runing eslint it adds taps and spaces , and it also change strings to template strings

Comment (mia-pi-git) [reply js/client-mainmenu.js:145]: > ther was issue when runing eslint it adds taps and spaces

Yes, that's proper behavior. It's supposed to do that to help format your code properly.

> t also change strings to template strings

I'd advise removing them after running eslint manually.

File: js/client-mainmenu.js (hunk 1 above)
Comment (mia-pi-git) [reply js/client-mainmenu.js:95]: Ah. It would be preferred to just use `app.on` and `app.trigger` instead.

File: js/client-mainmenu.js [hunk 10]
```diff
@@ -54,8 +54,8 @@
 			buf += '<p><button class="button mainmenu3" name="joinRoom" value="ladder">Ladder</button></p></div>';
 
 			buf += '<div class="menugroup"><p><button class="button mainmenu4 onlineonly disabled" name="joinRoom" value="battles">Watch a battle</button></p>';
-			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p></div>';
-
+			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p>';
+			buf += "<p class=link-generatore><button id='challeng-via-link' class='button mainmenu5 onlineonly'>send challeng via link</button></p></div>";
```
Comment (KrisXV) [inline js/client-mainmenu.js:58]:
```suggestion
			buf += '<p class="link-generator"><button id="challenge-via-link" class="button mainmenu5 onlineonly">Send challenge via URL</button></p></div>';
```

**Review by KrisXV — CHANGES_REQUESTED** [review]

**Comment by karamnaser** [comment]
I attached some vedios about the feature can you check if   the feature work as intended in game?

File: js/client-mainmenu.js [hunk 11]
```diff
@@ -114,7 +132,140 @@
...
+				window.Storage.prefs("errormassege",msg);
+				//validate team after receving validation messege from server when sending team
+				if(window.Storage.prefs("validateTeam")=="customteam"){
+					self.validateChallengeAndSend(self,{userId:window.Storage.prefs("challenger"),massage:msg});
+				}
+			}
+		},
+		generateLink:function(challengeData){
+			if(!$('.link').length){
+			let link="https://"+window.Config.routes.client+"?challenge-"+challengeData.userId+"-"+challengeData.format+"-"+challengeData.teamIndex;
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:205]: Why declare this twice? You never change it, so why not just declare it above the if statement?

File: js/client-mainmenu.js [hunk 12]
```diff
@@ -1487,5 +1638,21 @@
 			return buf;
 		}
 	});
-
+	window.addEventListener("load",function(){
+		if(window.location.search.includes("challenge")){
+			window.Storage.prefs("page","challenge");
+			setTimeout(function(){
+			//changing url without refreshing page, ther was issue of losing data when refreshing page 
+			window.history.replaceState(null,"","https://"+window.Config.routes.client+"/");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:1646]: Don't hardcode `https:`, use `location.protocol`.

File: js/client.js [hunk 13]
```diff
@@ -913,6 +913,8 @@ function toId() {
 		 * Receive from sim server
 		 */
 		receive: function (data) {
+			//trigger popup lisener when messege is received from server and save to storeg.pref obj  
+			app.trigger("lisenToPopUpMessege",{msg:data});
```
Comment (mia-pi-git) [inline js/client.js:917]: This is not necessarily accurate - this can be things other than a popup. That seems unintentional. I'd wait to trigger this until the `switch` statement later down that checks message types. Just call it in the `popup` case.

File: js/client-mainmenu.js [hunk 14]
```diff
@@ -114,7 +132,140 @@
...
+			let link="https://"+window.Config.routes.client+"?challenge-"+challengeData.userId+"-"+challengeData.format+"-"+challengeData.teamIndex;
+			$('.link-generatore').append("<p class=link>"+link+"</p>");
+			}
+			else{
+				let link="https://"+window.Config.routes.client+"?challenge-"+challengeData.userId+"-"+challengeData.format+"-"+challengeData.teamIndex;
+				$('.link').html("<p class=link>"+link+"</p>")
+			}
+		},
+		generateChallengeForm:function(self){
+			var buf = '<div class="challenge"><form class="battleform"><p>Challenge link generatore</p>';
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:214]:
```suggestion
			var buf = '<div class="challenge"><form class="battleform"><p>Challenge link generator</p>';
```

**Review by mia-pi-git — CHANGES_REQUESTED** [review]
Please run eslint.

**Comment by karamnaser** [comment]
done the require changes , made some changes on how challenge is triggered via link

File: js/client-mainmenu.js [hunk 15]
```diff
@@ -81,6 +81,17 @@
 
 			var self = this;
 			Storage.whenPrefsLoaded(function () {
+			//user log in observer that trigger challenge form when use log in
+				self.createLoginFormObserver(self);
+				$('#challeng-via-link').click(function () {
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:86]:
```suggestion
				$('#challenge-via-link').click(function () {
```

File: js/client-mainmenu.js [hunk 16]
```diff
@@ -81,6 +81,17 @@
...
+					if ($('.username').length) {
+						if (!$('.challenge').length) {
+							self.generateChallengeForm(self);
+						}
+					} else {
+						window.app.addPopupMessage("you need to log in first to generate link");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:92]:
```suggestion
						app.addPopupMessage("You need to be logged in to generate a challenge link.");
```

File: js/client-mainmenu.js [hunk 17]
```diff
@@ -114,7 +125,68 @@
...
+			const config = {attributes: true, childList: true, subtree: true};
+			const observer = new MutationObserver(function () {
+				if (window.app.user.get("named") && window.Storage.prefs("page") == "challenge") {
+					window.Storage.prefs("page", "testclient");
+					if ($('.challenge-container').length == 0) {
+						//user log in event trigerd when user log in and make user pick a team
+						$('body').on("login", function () {
+							self.onUserLogIn(self);
+						});
+						$('body').trigger("login");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:138]: Why do you add a listener and then immediately trigger it? Why not just call the function directly?

File: js/client-mainmenu.js [hunk 18]
```diff
@@ -114,7 +125,68 @@
...
+		onUserLogIn:function (self) {
+			let userId = window.Storage.prefs("challenger");
+			let format = window.Storage.prefs("format");
+			app.rooms[''].challenge(userId, format, (format.includes("random") ? '' : format));
+			$('.select.formatselect').prop("disabled",true);
+		},
+		generateLink:function (userId, format) {
+			if (!$('.link').length) {
+				let link = window.location.protocol+"//" + window.Config.routes.client + "?challenge-" + userId + "-" + format;
+				$('.link-generatore').append("<p class=link>" + link + "</p>");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:153]:
```suggestion
				$('.link-generator').append("<p class=link>" + link + "</p>");
```

File: js/client-mainmenu.js [hunk 19]
```diff
@@ -114,7 +125,68 @@
...
+			} else {
+				let link = window.location.protocol+"//" + window.Config.routes.client + "?challenge-" + userId + "-" + format;
+				$('.link').html("<p class=link>" + link + "</p>");
+				$('#generate-link').html("generate link");
+			}
+		},
+		generateChallengeForm:function (self) {
+			var buf = '<div class="challenge"><form class="battleform"><p>Challenge link generatore</p>';
+			buf += '<p><label class="label">Format:</label>' + self.renderFormats("gen8randombattle") + '</p>';
+			buf += '<button type="button" id=generate-link>generate link</button>';
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:164]:
```suggestion
			buf += '<button type="button" id=generate-link>Generate link</button>';
```

File: js/client-mainmenu.js [hunk 20]
```diff
@@ -114,7 +125,68 @@
...
+			buf += '<button type="button" id=close-challenge>close</button></form></div>';
+			$('.pmbox').append("<div class=challenge-container>" + buf + "</div>");
+			$('#generate-link').click(function () {
+				$('#generate-link').html("generating...");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:168]:
```suggestion
				$('#generate-link').html("Generating...");
```

File: js/client-mainmenu.js [hunk 21]
```diff
@@ -114,7 +125,68 @@
...
+				let link = window.location.protocol+"//" + window.Config.routes.client + "?challenge-" + userId + "-" + format;
+				$('.link').html("<p class=link>" + link + "</p>");
+				$('#generate-link').html("generate link");
+			}
+		},
+		generateChallengeForm:function (self) {
+			var buf = '<div class="challenge"><form class="battleform"><p>Challenge link generatore</p>';
+			buf += '<p><label class="label">Format:</label>' + self.renderFormats("gen8randombattle") + '</p>';
+			buf += '<button type="button" id=generate-link>generate link</button>';
+			buf += '<button type="button" id=close-challenge>close</button></form></div>';
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:165]:
```suggestion
			buf += '<button type="button" id=close-challenge>close</button></form></div>';
```
Element ids are bad practice in this project - you should be using the element's name and then checking which element it is in jQuery event listeners that are set on it.

File: js/client-mainmenu.js [hunk 22]
```diff
@@ -114,7 +125,68 @@
 				);
 			}
 		},
+		createLoginFormObserver:function (self) {
+			const config = {attributes: true, childList: true, subtree: true};
+			const observer = new MutationObserver(function () {
+				if (window.app.user.get("named") && window.Storage.prefs("page") == "challenge") {
+					window.Storage.prefs("page", "testclient");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:132]: ?

File: js/client-mainmenu.js [hunk 23]
```diff
@@ -114,7 +125,68 @@
 				);
 			}
 		},
+		createLoginFormObserver:function (self) {
+			const config = {attributes: true, childList: true, subtree: true};
+			const observer = new MutationObserver(function () {
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:130]: Why is this necessary? Can you not just use jQuery?

File: js/client-mainmenu.js [hunk 24]
```diff
@@ -54,8 +54,8 @@
 			buf += '<p><button class="button mainmenu3" name="joinRoom" value="ladder">Ladder</button></p></div>';
 
 			buf += '<div class="menugroup"><p><button class="button mainmenu4 onlineonly disabled" name="joinRoom" value="battles">Watch a battle</button></p>';
-			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p></div>';
-
+			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p>';
+			buf += '<p class=link-generatore><button id="challeng-via-link" class="button mainmenu5 onlineonly">send challenge via link</button></p></div>';
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:58]:
```suggestion
			buf += '<p class=link-generator><button class="button mainmenu5 onlineonly">Create challenge link</button></p></div>';
```
Button ids not preferred.

File: js/client-mainmenu.js [hunk 25]
```diff
@@ -114,7 +125,68 @@
...
+		},
+		generateLink:function (userId, format) {
+			if (!$('.link').length) {
+				let link = window.location.protocol+"//" + window.Config.routes.client + "?challenge-" + userId + "-" + format;
+				$('.link-generatore').append("<p class=link>" + link + "</p>");
+				$('#generate-link').html("generate link");
+			} else {
+				let link = window.location.protocol+"//" + window.Config.routes.client + "?challenge-" + userId + "-" + format;
+				$('.link').html("<p class=link>" + link + "</p>");
+				$('#generate-link').html("generate link");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:158]:
```suggestion
				$('#generate-link').html("generate link");
```
This should be using element names, not ids.

File: js/client-mainmenu.js [hunk 26]
```diff
@@ -114,7 +125,68 @@
...
+			let userId = window.Storage.prefs("challenger");
+			let format = window.Storage.prefs("format");
+			app.rooms[''].challenge(userId, format, (format.includes("random") ? '' : format));
+			$('.select.formatselect').prop("disabled",true);
+		},
+		generateLink:function (userId, format) {
+			if (!$('.link').length) {
+				let link = window.location.protocol+"//" + window.Config.routes.client + "?challenge-" + userId + "-" + format;
+				$('.link-generatore').append("<p class=link>" + link + "</p>");
+				$('#generate-link').html("generate link");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:154]:
```suggestion
				$('#generate-link').html("generate link");
```
This should be using element names, not ids.

File: js/client-mainmenu.js [hunk 27]
```diff
@@ -1487,5 +1559,20 @@
 			return buf;
 		}
 	});
-
+	window.addEventListener("load", function () {
+		if (window.location.search.includes("challenge")) {
+			window.Storage.prefs("page", "challenge");
+			setTimeout(function () {
+			//changing url without refreshing page, ther was issue of losing data when refreshing page
+				window.history.replaceState(null, "", window.location.protocol+"//" + window.Config.routes.client+"/");
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:1567]:
```suggestion
				window.history.replaceState(null, "", window.location.protocol + "//" + window.Config.routes.client + "/");
```

File: js/client-mainmenu.js [hunk 28]
```diff
@@ -1487,5 +1559,20 @@
 			return buf;
 		}
 	});
-
+	window.addEventListener("load", function () {
+		if (window.location.search.includes("challenge")) {
+			window.Storage.prefs("page", "challenge");
+			setTimeout(function () {
+			//changing url without refreshing page, ther was issue of losing data when refreshing page
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:1566]:
```suggestion
				// changing url without refreshing page, ther was issue of losing data when refreshing page
```

File: js/client-mainmenu.js [hunk 29]
```diff
@@ -1487,5 +1559,20 @@
...
+				window.history.replaceState(null, "", window.location.protocol+"//" + window.Config.routes.client+"/");
+				//log in user before sending challenge
+				new window.UserPopup({name:''}).login();
+			}, 1000);
+			let data = window.location.search.split("-");
+			let userId = data[1];
+			let format = data[2];
+			window.Storage.prefs("challenger", userId);
```
Comment (mia-pi-git) [inline js/client-mainmenu.js:1574]: Why is Storage.prefs necessary? This data shouldn't be needed across several sessions.

**Review by mia-pi-git — CHANGES_REQUESTED** [review]

File: js/client-mainmenu.js (hunk 22 above)
Comment (karamnaser) [reply js/client-mainmenu.js:132]: the code check when the user log in from the link and then activate the user login event

File: js/client-mainmenu.js (hunk 23 above)
Comment (mia-pi-git) [reply js/client-mainmenu.js:130]: Yeah, but I'm asking if you can't just use jQuery to watch the DOM instead? Or why not set a listener trigger in the code that handles completing logins?

File: js/client-mainmenu.js (hunk 22 above)
Comment (mia-pi-git) [reply js/client-mainmenu.js:132]: I'm asking what's with the `Storage.prefs('page')` call. It doesn't appear to be needed? If it is - why is it hardcoded?

Comment (karamnaser) [reply js/client-mainmenu.js:132]: there was an issue that trigger challenge after user log in from normal game link and not from the challenge link
so i needed to check if the user logged in after inserting challenge link and not normal link

Comment (mia-pi-git) [reply js/client-mainmenu.js:132]: Could you not just check the URL for that information? The `challenge-[]-[]` string should still be there and would be useful for that.

**Comment by karamnaser** [comment]
changed observer to app.once(login),removed id from html component ,removed prefstoreg instance and replaced them with 
obj in global scop,fixed spilling errors

File: js/client-mainmenu.js [hunk 30]
```diff
@@ -1,4 +1,9 @@
 (function ($) {
+	var challengeData = {
+	   userId :null,
+	   format : null,
+	   isChallengeTriggerd:false
+	};
```
Comment (AnnikaCodes) [inline js/client-mainmenu.js:6]:
```suggestion
	var challengeData = {
	   userId: null,
	   format: null,
	   isChallengeTriggered: false,
	};
```

File: js/client-mainmenu.js [hunk 31]
```diff
@@ -54,8 +59,8 @@
 			buf += '<p><button class="button mainmenu3" name="joinRoom" value="ladder">Ladder</button></p></div>';
 
 			buf += '<div class="menugroup"><p><button class="button mainmenu4 onlineonly disabled" name="joinRoom" value="battles">Watch a battle</button></p>';
-			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p></div>';
-
+			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p>';
+			buf += '<p class=link-generator><button class="challenge-via-link button mainmenu5 onlineonly">send challenge via link</button></p></div>';
```
Comment (AnnikaCodes) [inline js/client-mainmenu.js:63]:
```suggestion
			buf += '<p class=link-generator><button class="challenge-via-link button mainmenu5 onlineonly">Send challenge via link</button></p></div>';
```

File: js/client-mainmenu.js [hunk 32]
```diff
@@ -81,6 +86,26 @@
 
 			var self = this;
 			Storage.whenPrefsLoaded(function () {
+			//user log in event that trigger challenge form when user log in
+				app.once('init:choosename', function () {
+					if (challengeData.isChallengeTriggerd) {
+						challengeData.isChallengeTriggerd = false;
```
Comment (AnnikaCodes) [inline js/client-mainmenu.js:92]:
```suggestion
					if (challengeData.isChallengeTriggered) {
						challengeData.isChallengeTriggered = false;
```

File: js/client-mainmenu.js [hunk 33]
```diff
@@ -1487,5 +1557,18 @@
 			return buf;
 		}
 	});
-
+	window.addEventListener("load", function () {
+		if (window.location.search.includes("challenge")) {
+			challengeData.isChallengeTriggerd = true;
+			setTimeout(function () {
+			//changing url without refreshing page, ther was issue of losing data when refreshing page
+				window.history.replaceState(null,"", window.location.protocol + "//" + window.Config.routes.client + "/");
```
Comment (AnnikaCodes) [inline js/client-mainmenu.js:1565]:
```suggestion
				window.history.replaceState(null, "", window.location.protocol + "//" + window.Config.routes.client + "/");
```

**Review by AnnikaCodes — CHANGES_REQUESTED** [review]
Have you tested this locally?

**Comment by monsanto** [comment]
Thanks for the PR but it doesn't seem like you have addressed further feedback so I will assume that you are no longer interested.
